onebusaway-api-webapp/src/main/java/org/onebusaway/api/impl/TripsConfigurationServiceClientFileImpl.java

```diff
@@ -0,0 +1,153 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.onebusaway.api.impl;
+
+import java.io.File;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.io.FileUtils;
+import org.codehaus.jackson.JsonFactory;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.type.TypeReference;
+import org.onebusaway.api.services.TripsConfigurationServiceClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+public class TripsConfigurationServiceClientFileImpl implements
+		TripsConfigurationServiceClient {
+
+	private static Logger _log = LoggerFactory
+			.getLogger(TripsConfigurationServiceClientFileImpl.class);
+	private HashMap<String, Object> _config = null;
+	
+	private boolean isLocal = true;
+	
+	private String configFile = "/var/lib/oba/config.json";
+
+	public void setConfigFile(String file) {
+		_log.info("config service override of configFile=" + file);
+		configFile = file;
+	}
+
+	public TripsConfigurationServiceClientFileImpl() {}
+	
+	public TripsConfigurationServiceClientFileImpl(String configFile) {
+		_log.info("config service using configFile=" + configFile);
+		this.configFile = configFile;
+	}
+	
+	@PostConstruct
+	private void checkConfig(){
+		if(getConfig() == null)
+			isLocal = false;
+	}
+	
+	
+	@Override
+	public URL buildUrl(String baseObject, String... params) throws Exception {
+		// TODO Auto-generated method stub
+		_log.info("empty buildUrl");
+		return null;
+	}
+
+	@Override
+	public void setConfigItem(String baseObject, String component, String key,
+			String value) throws Exception {
+		// TODO Auto-generated method stub
+		_log.info("empty setConfigItem");
+	}
+
+	@Override
+	public String log(String baseObject, String component, Integer priority,
+			String message) {
+		// TODO Auto-generated method stub
+		_log.info("empty log");
+		return null;
+	}
+
+	@Override
+	public List<JsonObject> getItemsForRequest(String baseObject,
+			String... params) throws Exception {
+		// TODO Auto-generated method stub
+		_log.info("empty getItemsForRequest");
+		return null;
+	}
+
+	@Override
+	public List<Map<String, String>> getItems(String baseObject,
+			String... params) throws Exception {
+		_log.debug("getItems(" + baseObject + ", " + params + ")");
+		return (List<Map<String, String>>) getConfig().get("config");
+	}
+
+	@Override
+	public String getItem(String component, String line, String direction, String old_trip) throws Exception {
```

Comment:
This is all cut-and-paste code because you chose not to extend an existing class.  Worse yet, its bad code you copied-and-pasted, that was awkward due to legacy reasons.